File: chapter11/java/designpattern/src/main/java/com/designpattern/WindowsOS.java
package com.designpattern;

/**
 * WindowsOS implements the bridge "Implementor" interface OperatingSystem
 *
 * @version 1.0.0
 * @Author Bridget
 * @license MIT
 */
public class WindowsOS implements OperatingSystem {

    @Override
    public void displayOSName() {
        System.out.println("window");
    }

    @Override
    public void displayOSVersion() {
        System.out.println("11");
    }

    @Override
    public void displayMemory() {
        System.out.println("64 GB RAM");
    }

    @Override
    public void start() {
        System.out.println("Start Window Operating System");
    }

    @Override
    public void shutdown() {
        System.out.println("Shutdown Window Operating System");
    }
}

File: chapter11/typescript/WindowsOS.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.WindowsOS = void 0;
/**
 * WindowsOS implements the bridge "implementation" interface OperatingSystem
 *
 * @Author Bridget
 */
class WindowsOS {
    displayOSName() {
        console.log("window");
    }
    displayOSVersion() {
        console.log("11");
    }
    displayMemory() {
        console.log("64 GB RAM");
    }
    start() {
        console.log("Start Window Operating System");
    }
    shutdown() {
        console.log("Shutdown Window Operating System");
    }
}
exports.WindowsOS = WindowsOS;
